File: src/syntheticExamples/classDelegation/JDGUI/ClassDelegationDecompiled/ClassDelegation.java
package syntheticExamples.classDelegation.JDGUI.ClassDelegationDecompiled;

import kotlin.Metadata;

@Metadata(mv = {2, 0, 0}, k = 2, xi = 82, d1 = {"\000\b\n\000\n\002\020\002\n\000\032\006\020\000\032\0020\001\006\002"}, d2 = {"main", "", "KotlinDecompiler"})
public final class ClassDelegation {
    public static final void main() {
        BaseImpl base = new BaseImpl("Hello from BaseImpl");
        Derived derived = new Derived(base);
        derived.printMessage();
    }
}

File: src/syntheticExamples/classDelegation/JDGUI/ClassDelegationJ2K/ClassDelegation.kt
package syntheticExamples.classDelegation.JDGUI.ClassDelegationJ2K

object ClassDelegation {
    fun main() {
        val base = BaseImpl("Hello from BaseImpl")
        val derived = Derived(base)
        derived.printMessage()
    }

    @JvmStatic
    fun main(args: Array<String>) {
        main()
    }
}